src/analyses/apron/relationAnalysis.apron.ml

```diff
@@ -30,14 +30,22 @@ module VarinfoMapping = struct
     v.vname ^ "->" ^  name_varinfo t
 end 
 
-module PointerType = struct 
+(*Alloc ghost variable need to be global invariants in multithreading*)
+module GlobalPointer = struct 
+  let varType  = !ptrdiffType 
+  let isGlobal = true
+end
+module LocalPointer = struct 
   let varType  = !ptrdiffType 
   let isGlobal = false
 end
```

Comment:
In spirit of #1350, `ptrdiffType` should not be read at the top level because it might not yet be initialized properly.

Reply:
Fixed with e79b7d2756115263190ff62bff001204cb1ef81a